fog/ingest/server/src/worker.rs

```diff
@@ -126,15 +134,32 @@ impl IngestWorker {
                                 log::warn!(logger, "Failed updating ledger db metrics: {}", err);
                             }
 
-                            // Get the timestamp for the block.
-                            let timestamp =
-                                watcher.poll_block_timestamp(next_block_index, watcher_timeout);
+                            // Tracing
+                            let tracer = tracer!();
+
+                            let mut span =
+                                block_span_builder(&tracer, "poll_block", next_block_index)
+                                    .with_start_time(start_time)
+                                    .start(&tracer);
 
-                            controller.process_next_block(
-                                block_data.block(),
-                                block_data.contents(),
-                                timestamp,
+                            span.set_attribute(
+                                TELEMETRY_BLOCK_INDEX_KEY.i64(next_block_index as i64),
                             );
+
+                            let _active = mark_span_as_active(span);
+
+                            // Get the timestamp for the block.
+                            let timestamp = tracer.in_span("poll_block_timestamp", |_cx| {
+                                watcher.poll_block_timestamp(next_block_index, watcher_timeout)
+                            });
+
+                            tracer.in_span("process_next_block", |_cx| {
```

Comment:
It seems that sometimes we're doing
```
let mut span = ...;
...
span.end()
```

And sometimes we do this thing where the code is stuffed in a lambda

Fwiw I find it a little more readable when the span is a separate statement, but I'm not gonna block the PR
Especially in the case where lambda returns something

They are actually equivalent though right? Or does one of them offer extra functionality?

Reply:
The `in_span` does not let you control aspects of the span you are creating like the trace id or start/end times, which is why it is only useful when you're already inside a span that has been initialized with the desired TraceId.